file.go

```diff
@@ -1,37 +1,204 @@
 package jcsv
 
-import "os"
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
 
 type file struct {
-	// TODO: define variables if needed
+	data []map[string]interface{}
 }
 
 func ParseJsonFile(path string) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	//open JSON file
+	jsonFile, err := os.Open(path)
+
+	if err != nil {
+		fmt.Println(err)
+		return file{}, err
+	}
+
+	defer jsonFile.Close()
+
+	//read complete json file and store in byteValue
+	byteValue, err := ioutil.ReadAll(jsonFile)
+
+	if err != nil {
+		fmt.Println(err)
+		return file{}, err
+	}
+
+	var myFile file
+	//convert JSON read from file in byteValue to "Data" inside file stucture
+	json.Unmarshal(byteValue, &myFile.data)
+
+	return myFile, err
 }
 
 func ParseOpenedJsonFile(f *os.File) (file, error) {
-	// TODO: read the given file into your `file` object
-	return file{}, nil
+	var myFile file
+
+	//read complete json file and store in byteValue
+	byteValue, err := ioutil.ReadAll(f)
+
+	if err != nil {
+		fmt.Println(err)
+		return file{}, nil
+	}
+
+	//convert JSON read from file in byteValue to "Data" inside file stucture
+	json.Unmarshal(byteValue, &myFile.data)
+
+	return myFile, err
 }
 
 func ParseCsvFile(path string, hasHeaders bool) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	//open CSV file
+	csvFile, err := os.Open(path)
+
+	if err != nil {
+		return file{}, err
+	}
+
+	defer csvFile.Close()
+
+	//get CSV Reader instance
+	csvReader := csv.NewReader(csvFile)
+	//create an object to hold csv data
+	var CSVData [][]string
+	//read all data into CSVData
+	CSVData, err = csvReader.ReadAll()
+	//if an error occurred
+	if err != nil {
+		return file{}, nil
+	}
+
+	var myFile file
+	myFile.data = make([]map[string]interface{}, len(CSVData))
```

Comment:
You won't need to do this.
Just append the `map[string]interface{}` to `myFile.data` later.